classes/Context.php

```diff
@@ -391,7 +392,7 @@ public function getTranslatorFromLocale($locale)
         $translator->addLoader('xlf', new XliffFileLoader());
 
         $sqlTranslationLoader = new SqlTranslationLoader();
-        if (!is_null($this->shop)) {
+        if (!is_null($this->shop) && $this->shop->theme instanceof Theme) {
```

Comment:
From the looks of the stack trace in [this comment by @marionf](https://github.com/PrestaShop/PrestaShop/pull/8843#issuecomment-374659710), this happens when the configuration for `PS_LANG_DEFAULT` is `NULL`: https://github.com/PrestaShop/PrestaShop/blob/4fb28a9dc9e12ccf4da2a48b8dade087383d58f1/classes/Language.php#L1342

Any idea why that can happen?
And what happens if `$this->shop->theme` is not an instance of `Theme` and we skip this line?

Reply:
The translations would not be loaded, which was okay if there was no theme anyway.

However, we should take some more time to check the stack trace from Marion, which gives us a new clue on what's wrong. My change does not fix the original issue, it could hide it instead.